parachain/primitives/core/src/rewards.rs

```diff
@@ -0,0 +1,31 @@
+use frame_support::traits::{Currency, ExistenceRequirement::KeepAlive};
+use frame_system::Config;
+use sp_runtime::traits::Zero;
+use sp_std::marker::PhantomData;
+
+
+pub trait RewardRelayer<AccountId, Balance> {
+	fn pay_relayer(source: &AccountId, relayer: &AccountId, reward: Balance);
+}
+
+pub struct InstantRewards<T, C>(PhantomData<(T, C)>);
+
+impl<T, C> RewardRelayer<T::AccountId, C::Balance> for InstantRewards<T, C>
```

Comment:
This is a thing that actually pays relayers. It pays them instantaneously per message.